### PR TITLE
Rename SURFBOARD to SURF_ITEM

### PR DESCRIPTION
**Pull request by SatoMew**
Likewise, rename `ItemUseSurfboard` to `ItemUseSurf`.

This PR also fixes some typos and inconsistencies, and cleans up a few comments.

### CODE REVIEW

**Comment by dannye** [comment]
I'm still not sold on the value of renaming `SURFBOARD` to `SURF_ITEM`.

All item constants are derived from their text name, except for the 2 items with the placeholder name of "?????", so for these two there is no objectively "right" name -- it's up to us. And I like Surfboard.

If we conceptualize the item as a physical object that the player could reasonably obtain and hold in their pack, surfboard makes sense. What would it mean to hold a "surf item" in your bag?

Functionally, the relationship between the surf move and the surf item is exactly the same as the relationship between the dig move and the escape rope item, with the only difference being that escape rope is a properly legitimate item with a real name and obtainable in-game. I just don't think the name of the item needs to be "abstracted" to the point of not referring to a real object.

Getting less important and more subjective, I think it's nice that the surfboard comes immediately after the bicycle in the items list. The two form a pair of the two objects in the game that enable the player to enhance their movement around the overworld. I think bicycle/surfboard are a better yin/yang than bicycle/surf item.

Overall, it's not the biggest deal either way, but I don't really want to make the change if I'm not truly compelled.

File: constants/item_constants.asm [hunk 1]
```diff
@@ -52,13 +52,13 @@ DEF SAFARI_ROCK EQU $16 ; overload
 	const DOME_FOSSIL   ; $29
 	const HELIX_FOSSIL  ; $2A
 	const SECRET_KEY    ; $2B
-	const UNUSED_ITEM   ; $2C "?????"
+	const DUMMY_ITEM    ; $2C ; unused
```
Comment (dannye) [inline constants/item_constants.asm:55]: What's the motivation for this change? I'm neither for or against the new name hypothetically, but again why introduce renames that will throw off old code if there isn't a compelling reason?

File: home/item.asm [hunk 2]
```diff
@@ -1,23 +1,7 @@
...
 	farjp UseItem_
 
-; confirms the item toss and then tosses the item
-; INPUT:
-; hl = address of inventory (either wNumBagItems or wNumBoxItems)
-; [wcf91] = item ID
-; [wWhichPokemon] = index of item within inventory
-; [wItemQuantity] = quantity to toss
-; OUTPUT:
-; clears carry flag if the item is tossed, sets carry flag if not
```
Comment (dannye) [inline home/item.asm:20]: Why delete these descriptions? Explaining input and output is nice actually.

File: constants/item_constants.asm (hunk 1 above)
Comment (dannye) [reply constants/item_constants.asm:55]: I don't mean the comment, I mean the constant. Why is `DUMMY_ITEM` better than `UNUSED_ITEM`?

File: home/item.asm (hunk 2 above)
Comment (dannye) [reply home/item.asm:20]: Comments which describe the input/output of a routine are not at all the type of redundant comments referred to by #302.

That being said, these description comments are also already repeated on `TossItem_`, `IsKeyItem_`, so removing them from here _could_ be fine, but the description on `UseItem` is not repeated on `UseItem_`.

That being said, it's kind of nice for the description to be on the entrypoint in home/ rather than the implementation in engine/, since the entrypoints are kind of like the "api" of home; convenient to access and convenient to read.

I really like the way this approach is used in poketcg for example:
https://github.com/pret/poketcg/blob/43b7b92/src/home/coin_toss.asm#L1-L13

The comments are not "misplaced" as you say; they are absolutely true in reference to the routines where they are placed.

File: constants/item_constants.asm (hunk 1 above)
Comment (dannye) [reply constants/item_constants.asm:55]: That seems like a poor reason to change the name.

**Comment by dannye** [comment]
> It's definitely not a surfboard though.

Going off of only the rom, what we have, objectively, is an item named "?????" which when used allows the player to surf on water without the help of a pokemon. What does this sound like? Sounds like a surfboard to me.

**Comment by dannye** [comment]
This patch feels like a mixed bag of changes for change's sake, which I've never been too keen on. Some of the changes are good, but most seem unnecessary.

File: constants/item_constants.asm (hunk 1 above)
Comment (dannye) [reply constants/item_constants.asm:55]: `UNUSED_ITEM` is good and doesn't need changing. We already use "UNUSED" in many other constant names, like all the unused maps and unused sprites, as well as a couple others.

File: home/item.asm (hunk 2 above)
Comment (dannye) [reply home/item.asm:20]: I could be persuaded both ways. I see pros and cons to both.

Taking the coin toss example from poketcg, the implementation routine has no description comment on top, which makes sense to me since the code more so speaks for itself:
https://github.com/pret/poketcg/blob/43b7b92/src/engine/duel/core.asm#L7817

Although maybe it would be better if the implementation routine at least had a tl;dr description or something, I dunno.

I'm also not super concerned with standardizing, given that many different people work on the different disassemblies, and the games themselves are structured differently enough that it may be best to consider many things on an individual level. Of course, consistency and uniformity when it's easy is good.

**Review by vulcandth — CHANGES_REQUESTED** [review]

**Comment by Rangi42** [comment]
I would actually take the opposite tack of this PR: leave `SURFBOARD` alone, and take the pokecrystal approach of renaming `UNUSED_ITEM` to `ITEM_2C` and `PP_UP_2` to `ITEM_32`.

(It's mainly the `PP_UP_2` name that bothers me: it's potentially confusing since `PP_UP` exists, and the item doesn't even have the `ItemUsePPUp` effect. It just shares the name, and for that we could just clarify with a comment: `li "PP UP" ; ITEM_32`. But then that means there are two "unused" items, and I would prefer to avoid the ordinal `UNUSED_ITEM_2`, so the Gen 2 approach unambiguously identifies it no matter how many unused items there are.)

File: constants/item_constants.asm (hunk 1 above)
Comment (vulcandth) [inline constants/item_constants.asm:55]:
```suggestion
	const ITEM_2C       ; $2C ; unused
```

**Review by vulcandth — CHANGES_REQUESTED** [review]

**Comment by dannye** [comment]
> I would actually take the opposite tack of this PR: leave `SURFBOARD` alone, and take the pokecrystal approach of renaming `UNUSED_ITEM` to `ITEM_2C` and `PP_UP_2` to `ITEM_32`.

I'm on board with that.
It's also worth noting that constants/map_constants.asm uses the convention of `UNUSED_MAP_0B`, `UNUSED_MAP_69`, etc, which I like.
So we could also go with `UNUSED_ITEM_2C` and `UNUSED_ITEM_32`. I'm fine either way.

**Comment by Rangi42** [comment]
`UNUSED_ITEM_XX` sounds good.

**Comment by vulcandth** [comment]
> `UNUSED_ITEM_XX` sounds good.

I mean.. i'm fine with it.. but its redundant with the `; unused` comment... which is what people will probably grep/search for when looking for unused things to delete.

**Comment by Rangi42** [comment]
Sure, `UNUSED_` or not.
